samcli/commands/publish/app/cli.py

```diff
@@ -0,0 +1,137 @@
+"""CLI command for "publish app" command."""
+
+import os
+import json
+import logging
+import click
+
+from botocore.exceptions import ClientError
+
+from serverlessrepo import publish_application
+from serverlessrepo.publish import CREATE_APPLICATION
+from serverlessrepo.exceptions import ServerlessRepoError
+
+from samcli.cli.main import pass_context, common_options as cli_framework_options, aws_creds_options
+from samcli.commands._utils.options import template_common_option
+from samcli.commands.local.cli_common.user_exceptions import SamTemplateNotFoundException, S3PermissionsRequired
+from samcli.commands.exceptions import UserException
+
+LOG = logging.getLogger(__name__)
+
+HELP_TEXT = """
+Use this command to publish a packaged AWS SAM template to
+the AWS Serverless Application Repository to share within your team,
+across your organization, or with the community at large.\n
+\b
+This command expects the template's Metadata section to contain an
+AWS::ServerlessRepo::Application section with application metadata
+for publishing. For more details on this metadata section, see
+https://docs.aws.amazon.com/serverlessrepo/latest/devguide/serverless-app-publishing-applications.html
+\b
+Examples
+--------
+To publish an application
+$ sam publish app -t packaged.yaml --region <region>
+"""
+SHORT_HELP = "Publish a packaged AWS SAM template to the AWS Serverless Application Repository."
+PUBLISH_GUIDE = "https://docs.aws.amazon.com/serverlessrepo/latest/devguide/serverless-app-publishing-applications.html"
+
+
+@click.command("app", help=HELP_TEXT, short_help=SHORT_HELP)
+@template_common_option
+@aws_creds_options
+@cli_framework_options
+@pass_context
+def cli(ctx, template):
+    # All logic must be implemented in the ``do_cli`` method. This helps with easy unit testing
+
+    do_cli(ctx, template)  # pragma: no cover
+
+
+def do_cli(ctx, template):
+    """Publish the application based on command line inputs."""
+    if not os.path.exists(template):
+        click.secho("Publish Failed", fg='red')
+        raise SamTemplateNotFoundException("Template at {} is not found".format(template))
+
+    with click.open_file(template, 'r') as template_file:
+        try:
+            output = publish_application(template_file.read())
```

Comment:
Wasn't there a change to the serverlessrepo module so you could pass in a client? How is the client getting initialized with the correct credentials?

Reply:
The region/profile is set through sam cli [context](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/cli/context.py#L77) and passed to every command annotated with [aws_creds_options](https://github.com/awslabs/aws-sam-cli/pull/852/files/cc69a37f1e533b6321e588ec1f889eb2f9475c46#diff-2237f386debbe0f92032b8e0ceda2ae5R42). The boto3 client is initialized through `boto3. setup_default_session` therefore passing a sar_client is unnecessary.